### PR TITLE
Improve fault status

### PR DESCRIPTION
**Pull request by labbott**

### CODE REVIEW

File: humility-arch-cortex/src/debug.rs [hunk 1]
```diff
@@ -135,31 +135,257 @@ register!(CPUID, 0xe000_ed00,
...
+    pub invep, _: 0;
+);
+
+impl std::fmt::Debug for SFSR {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.lserr() {
+            write!(f, "Lazy state error")?;
+        }
+        if self.lsperr() {
+            write!(f, "Error preserving floting point state")?;
```
Comment (cbiffle) [inline humility-arch-cortex/src/debug.rs:157]: If more than one bit is set here, the sentences will concatenate without intervening whitespace.

**Review by cbiffle — COMMENTED** [review]
This is an odd request, but: could you please print the literal hex value of the CFSR/SFSR. This will make it easier to compare to the architecture spec and remove the possibility of us failing to explain any bits. For instance, afaict, if UFSR has bit 5/6/7 set, this will silently fail to acknowledge that right now.

**Comment by labbott** [comment]
```
humility:           PC => 0x36c      
humility:          PSR => 0x81000007 
humility:          MSP => 0x20003298 
humility:          PSP => 0x0        
humility:          SPR => 0x0        
humility: Secure Fault @ 3000000 : (raw SFSR 0x48) Access violation
```

Sample output

File: humility-arch-cortex/src/debug.rs (hunk 1 above)
Comment (labbott) [reply humility-arch-cortex/src/debug.rs]: Added a space to the end of sentences

**Comment by labbott** [comment]
Another sample output:

```
humility:           PC => 0x5ecc     
humility:          PSR => 0x69000003 
humility:          MSP => 0x1ffffb80 
humility:          PSP => 0x0        
humility:          SPR => 0x0        
humility: Fault detected! Raw CFSR: 0x9200
humility: Bus Fault @ 1ffffb90 : Precies data access error Bus Error during exception entry stacking
```

**Review by cbiffle — APPROVED** [review]
Nit: "Precise" is mispeld.

This is better than what we've got, I'm personally comfortable with you merging it and improving the reporting later if you still feel the need.

**Pull request merged**